### PR TITLE
[13.0][IMP]bi_sql_editor: many2one fields clickable on the list view if setting is enabled

### PR DESCRIPTION
**Pull request by GuillemCForgeFlow**
cc @ForgeFlow

### CODE REVIEW

**Comment by GuillemCForgeFlow** [comment]
@legalsylvain perhaps you can give me some feedback on this one as well 🙂

File: bi_sql_editor/views/view_bi_sql_view.xml [hunk 1]
```diff
@@ -178,6 +178,10 @@ License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
                                         name="is_group_by"
                                         attrs="{'invisible': [('field_description', '=', False)]}"
                                     />
+                                    <field
+                                        name="is_many2one_clickable"
```
Comment (MiquelRForgeFlow) [inline bi_sql_editor/views/view_bi_sql_view.xml]: I would add this after many2one_model_id, as it's related to many2one field type too.

Comment (GuillemCForgeFlow) [reply bi_sql_editor/views/view_bi_sql_view.xml]: I have to say that sometimes you are right, and this is one of those few times. thank you for the input 🙂

**Review by MiquelRForgeFlow — APPROVED** [review]

**Review by BernatPForgeFlow — APPROVED** [review]
Functional test on Guillem's computer :computer: :slightly_smiling_face:

**Comment by GuillemCForgeFlow** [comment]
@legalsylvain it would be nice to also get your input on this feature whenever you can 🙂

**Comment by legalsylvain** [comment]
Hi thanks for the ping ! 

In fact, I don't understand what is widget="many2one". I rgrepped Odoo, and there is only a fiew occurences in the whole project. Coud you elaborate what is changing ?

**Comment by GuillemCForgeFlow** [comment]
> Hi thanks for the ping !
> 
> In fact, I don't understand what is widget="many2one". I rgrepped Odoo, and there is only a fiew occurences in the whole project. Coud you elaborate what is changing ?

Sure, it is actually a widget added in odoo's core. See here: https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/fields/relational_fields.js#L96 and here https://github.com/odoo/odoo/blob/13.0/addons/web/static/src/js/fields/field_registry.js#L72.

It's not widely used, but it gives the possibility for Many2one records to be clickable so you can access the specific record.

**Review by legalsylvain — CHANGES_REQUESTED** [review]
Well, I just tested on runboat : 

**checkbox disabled :**
 
image

**checkbox enabled :** 

image

(Add possibility to click on the company to enter in the company form).


Well, I'm not very in favor of adding another extra setting (extra column in the configuration of the view of bi.sql.view.field model). 

Why not just install the OCA/web module web_tree_many2one_clickable
(https://github.com/OCA/web/tree/16.0/web_tree_many2one_clickable)

What do you think ? 

thanks !

**Comment by GuillemCForgeFlow** [comment]
> Well, I just tested on runboat :
> 
> **checkbox disabled :**
> 
> image
> 
> **checkbox enabled :**
> 
> image
> 
> (Add possibility to click on the company to enter in the company form).
> 
> Well, I'm not very in favor of adding another extra setting (extra column in the configuration of the view of bi.sql.view.field model).
> 
> Why not just install the OCA/web module web_tree_many2one_clickable (https://github.com/OCA/web/tree/16.0/web_tree_many2one_clickable)
> 
> What do you think ?
> 
> thanks !

Understandable, I agree we can still use the https://github.com/OCA/web/tree/16.0/web_tree_many2one_clickable module if needed. We can close it 👍🏿

**Pull request closed without merging**